docs/did/asset-did-spec.md

```diff
@@ -0,0 +1,107 @@
+[![](../../.maintain/media/kilt-header.png)](https://kilt.io)
+
+# Asset Decentralized Identifiers (DID) Method Specification
+
+### Editors
+
+- **Antonio Antonino** - KILT Protocol [antonio@kilt.io](mailto:antonio@kilt.io)
+
+### Version History
+
+- **v1.0 - July.04 2022**: Initial definition and RFC for the new method
+
+---
+
+## Abstract
+
+This document defines the Asset DID Method that conforms to the [DID Core W3C Spec][did-core-spec].
+By the term *asset*, the authors refer to both an entire asset (or token) class (e.g., the entire Bitcoin fungible asset class) as well as a specific instance within that class (e.g., a specific NFT - the instance - within an NFT collection - the asset class).
+
+Hence, an Asset Decentralized Identifier (DID) is a string uniquely identifying assets on any existing and future blockchain.
+
+## Method Syntax
+
+### DID Scheme and Identifiers
+
+The Asset DID method is identified by the `asset` scheme, as in the following example: `did:asset:bip122:000000000019d6689c085ae165831e93.slip44:0`.
+
+Asset DIDs are composed of the following elements:
+
+```
+asset-did                   = "did:asset:" + <modified-caip-19-asset-id>
+modified-caip-19-asset-id   = <caip-2-chain-id> + "." + <asset-namespace> + ":" + <asset-reference> + [":" + <asset-id>]
+caip-2-chain-id             = <chain-namespace> + ":" + <chain-reference>
+chain-namespace             = [-a-z0-9]{3,8}
+chain-reference             = [-a-zA-Z0-9]{1,32}
+asset-namespace             = [-a-z0-9]{3,8}
+asset-reference             = [-a-zA-Z0-9]{1,64}
+asset-id                    = [-a-zA-Z0-9]{1,78}
+```
+
+The `caip-2-chain-id` is a CAIP (Chain Agnostic Improvement Proposal) chain identifier as described in the [CAIP-2 Blockchain ID Specification][caip-2-spec].
+The `modified-caip-19-asset-id` is derived from a CAIP asset identifier as described in the [CAIP-19 Asset Type and Asset ID Specification][caip-19-spec], with the `/` symbol separating the chain identifier from the asset identifier being replaced with the `.` symbol, and the optional `/` symbol preceding the `asset-id` component being replaced by the `:` symbol.
```

Comment:
The point is exactly readability. It is not nice, in my opinion, to have percent-encoded characters in a DID. Plus, other similar DID methods like the [NFT DID](https://github.com/ceramicnetwork/CIP/blob/main/CIPs/CIP-94/CIP-94.md) do something similar, but using a different character for replacement.

Reply:
I would also vote for the custom conversion, since percentage encoding (at least in the URL context) is rather applied on query fragments and not on the domain part. I.e. every did would include percentage encoded strings, not just some.

Reply:
ok, that's fair